mautic/mautic
# Fix: GrapesJS assets transfer is slowing admin panel view loading

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ Y ]
| New feature/enhancement? (use the a.x branch)      | [ N ]
| Deprecations?                          | [ N ]
| BC breaks? (use the c.x branch)        | [ N ]
| Automated tests included?              | [ Y ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
I noticed that all GrapesJS assets are sent with view requests. In this case, if you have a lot of assets, it can significantly slow down view loading.

In my case, with large assets database the Contact, Companies, Segments, Campaigns views are transferring > 2MB of GrapesJS data.
![image](https://user-images.githubusercontent.com/8580942/207875667-c9f8570a-2e72-4c1a-95ed-08c78830df34.png)
![image](https://user-images.githubusercontent.com/8580942/207881390-a5799051-28f5-4677-ab02-76ed86b268ec.png)


This PR changes this behavior and loads assets asynchronously when GrapesJS builder is initialized. 



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open ten test GrapesJS builder including assets 
3. GrapesJS assets should not be transferred in panel view requests as encoded JSON, it can be verified using browser dev tools:
![image](https://user-images.githubusercontent.com/8580942/207880019-9011f80d-0a62-48a4-936c-1437635d261d.png)
